### ISSUE TITLE
Idea is returned from search twice if it has 2 similarly named tags

### ISSUE THREAD

**Issue by Mctalian** [YashdalfTheGray/flintandsteel#230]
I thought this was a bigger deal, but I think ng-repeat handles this. It throws a console error about duplicates in the ng-repeat. But anyway, this test case would fail if we had it in there:

`searchFilter.spec.js`
```javascript
it('should only have an idea in the results once', function() {
    var query = "TestTag";
    expect(search(testList, query).length).toEqual(4);
});
```
```
PhantomJS 1.9.8 (Windows 8 0.0.0) SearchFilter should only have an idea in the results once FAILED
        Expected 6 to equal 4.
            at C:/Users/Bobby/Documents/GitHub/flintandsteel/src/ideas/searchFilter/searchFilter.spec.js:138
PhantomJS 1.9.8 (Windows 8 0.0.0): Executed 130 of 130 (1 FAILED) (1.343 secs / 1.551 secs)
[22:41:57] 'test:client' errored after 8.99 s
```

**Comment by Mctalian** [comment]
eh @Daviddt17, I don't even think the ng-repeat complains about duplicates. So you're free to just close this if you want. I don't think it adds any value. You can work on the other thing we discussed, the Events Page.

**Comment by LongArmMcGee** [comment]
We discussed this during our Innovation Challenge Committee meeting and decided it's not very important. We'll do it with the other issues after MVP. Bob will find something more productive for you to work on.

**Comment by Daviddt17** [comment]
I was just working on this haha. Sorry I'm at home today. I'm pretty sure my solution is only like 3 lines. Should I just forget it?

**Comment by Daviddt17** [comment]
We might have a new issue. I was just testing out searching and it doesn't work for tags or events yet? I'm pretty sure I had searching for tags working at one point. The tests work because we made the test objects with all the necessary data. I assume something broke when Yash made the REST changes? I'm not sure since I've been away from this project for a while.

**Comment by Mctalian** [comment]
If you already have a solution we can put up A PR I think. But we'll have
to talk about the searching thing, I swear I had event searching working at
some point

**Comment by Daviddt17** [comment]
My bad, it does work for events!

**Comment by Daviddt17** [comment]
Still nothing for tags. I remember having to change back-end code so that the object we get back also gave us tags, but that was when Yash was making his changes so the things I added might have been wiped out. Anyway I just pushed the solution for this issue so I'll make a PR.

**Comment by Mctalian** [comment]
I'm not certain searching for tags ever actually worked after looking through the history. It looks like you added tags to the projection on the server side, but never added it as a key-value pair before pushing it to the header array.

[Check within the `push` call at line 64 in `\server\ideas\ideas.js`](https://github.com/YashdalfTheGray/flintandsteel/blob/master/server/ideas/ideas.js#L64). Adding tags to that should fix it. If you can add that to your PR we'll be all good I think.

**Comment by Daviddt17** [comment]
Resolved with #260
